src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -158,8 +228,30 @@ void java_bytecode_convert_classt::convert(const classt &c)
   // interfaces are recorded as bases
   for(const auto &interface : c.implements)
   {
-    symbol_typet base("java::"+id2string(interface));
-    class_type.add_base(base);
+    const symbol_typet base("java::" + id2string(interface));
+
+    // if the interface is generic then the class has the interface reference
+    // including the generic info in its signature
+    // e.g., signature for class 'A' that implements
+    // 'GenericInterface<Integer>' is 'Ljava/lang/Object;
+    // LGenericInterface<LInteger;>;'
+    if(
+      c.signature.has_value() &&
+      !extract_generic_interface_reference(
```

Comment:
Rather than calling the function twice it is probably clearer to store the result. It might also simplify things if you shift the check for `c.signature.has_value()`  inside the extraction method